mutiny-core/src/lib.rs

```diff
@@ -831,9 +831,8 @@ impl<S: MutinyStorage> MutinyWallet<S> {
             None
         } else {
             self.create_lightning_invoice(amount, labels.clone())
-                .await
-                .ok()
-                .and_then(|invoice| invoice.bolt11)
+                .await?
+                .bolt11
```

Comment:
this was the code before: 

```
        // If we are in safe mode, we don't create invoices
        let invoice = if self.safe_mode {
            None
        } else {
            let inv = self.create_invoice(amount, labels.clone()).await?;
            Some(inv.bolt11.ok_or(MutinyError::WalletOperationFailed)?)
        };
```

needs an `ok_or` and wrapped in `Some` since it is an option

Reply:
Brought back, made the error `InvoiceCreationFailed` because that seemed more correct

Reply:
Well, is the frontend parsing the error and deciding which one to do?

Reply:
So we just display it

Reply:
No, it does switch on the error type, but in this case not this specific error so it's fine. 

```js
        } catch (e) {
            console.error(e);
            if (e === "Satoshi amount is invalid") {
                setError(i18n.t("receive.error_under_min_lightning"));
            } else {
                setError(i18n.t("receive.error_creating_unified"));
            }
        }
```

Reply:
That is for checking the min lightning amount, I thought we displayed this error in the toast message

Reply:
No, this code is based on the return of this API call. It parses the error we return here and it is not taking it raw.